core/capabilities/remote/target/request/client_request.go

```diff
@@ -102,6 +111,7 @@ func (c *ClientRequest) Expired() bool {
 }
 
 func (c *ClientRequest) Cancel(err error) {
+	c.cancelFn()
```

Comment:
Do we need to wait here too?
```suggestion
	c.cancelFn()
	c.wg.Wait()
```

Reply:
it does indeed, added that